python310Packages.attrs: 22.1.0 -> 22.2.0

tjni opened the pull request:
###### Description of changes

https://github.com/python-attrs/attrs/releases/tag/22.2.0

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

dotlambda committed:
python3Packages.pythonImportsCheck: set $PYTHONPATH

Don't rely on the installPhase doing so.

dotlambda committed:
python310Packages.pytest-xdist: 2.5.0 -> 3.1.0

https://github.com/pytest-dev/pytest-xdist/blob/v3.1.0/CHANGELOG.rst

bbjubjub2494 committed:
python3Packages.typing-extensions: 4.3.0 -> 4.4.0

dotlambda committed:
python310Packages.pytest-forked: add setup hook

tjni committed:
python310Packages.attrs: 22.1.0 -> 22.2.0

mweinelt committed:
python39Packages: Stop recursing into package set

For NixOS 23.05 we are shifting support away from 3.9 and towards 3.10
and 3.11.

mweinelt committed:
python311Packages: Recurse into package set

Python 3.11 is the latest stable release that we are going to support,
together with 3.10, in NixOS 23.05. The default python version will
remain at 3.10, but 3.11 packages will be built on Hydra and provided
from the binary cache.

mweinelt committed:
python311Packages.tensorflow-bin: Disable

The upstream does not support 3.11 yet and as such has not provided any
compatible wheels.

mweinelt committed:
python3Packages.setuptools: 65.3.0 -> 65.6.3

https://setuptools.pypa.io/en/stable/history.html#v65-6-3

mweinelt committed:
python3Packages.pip: 22.2.2 -> 22.3.1

https://pip.pypa.io/en/stable/news/#v22-3-1

mweinelt committed:
python3Packages.hypothesis: 6.54.5 -> 6.61.0

https://hypothesis.readthedocs.io/en/latest/changes.html#v6-61-0

mweinelt committed:
python3Packages.flit-scm: Trim dependencies and refactor

mweinelt committed:
python3Packages.exceptiongroup: Disable tests on python310

Since pytest now requires exceptiongroup we can't use it to test the
package, because it causes an infinite recursion.

mweinelt committed:
python3Packages.pytest: 7.1.3 -> 7.2.0

https://github.com/pytest-dev/pytest/releases/tag/7.2.0

mweinelt commented on pkgs/development/python-modules/attrs/default.nix:
```diff
@@ -7,12 +7,12 @@
-  version = "22.1.0";
-  disabled = pythonOlder "3.5";
+  version = "22.2.0";
+  disabled = pythonOlder "3.6";
```
```suggestion
  disabled = pythonOlder "3.6";
  format = "pyproject";
```

mweinelt commented:
The tests derivation requires a format to be set.

mweinelt committed:
python3Packages.pytest-mock: 3.8.2 -> 3.10.0

https://github.com/pytest-dev/pytest-mock/releases/tag/v3.10.0
https://github.com/pytest-dev/pytest-mock/releases/tag/v3.9.0

mweinelt committed:
python3Packages.cffi: Apply patches for pytest 7.2.0 compat

mweinelt committed:
python3Packages.pytest-xprocess: 0.20.0 -> 0.21.0

https://github.com/pytest-dev/pytest-xprocess/blob/0.21.0/CHANGELOG.rst

mweinelt committed:
python3Packages.execnet: Fix compat with pytest 7.2.0

mweinelt committed:
python3Packages.flit: 3.7.1 -> 3.8.0

https://flit.pypa.io/en/stable/history.html#version-3-8

tjni commented:
Thanks for the help, guys. I have updated the PR addressing the review comments. I kept the `format` as `other` in the test derivation, since using `pyproject` didn't work (admittedly, I didn't try very hard -- only tried copying `pyproject.toml` over).

mweinelt commented:
Cherry-picked into python-updates. Thanks!
